.github/workflows/promote-staging-to-prod.yml

```diff
@@ -0,0 +1,77 @@
+name: Promote Staging to Prod
+
+on:
+  workflow_dispatch: {}
+  repository_dispatch:
+    types: [deployment]
+
+jobs:
+  check_staging_commit_checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: staging
+      - name: Check if Latest Staging Post-Deploy Tests are Passing
+        id: get_checks
+        run: |
+          checks=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+                       "https://api.github.com/repos/chanzuckerberg/single-cell-data-portal/actions/runs?actor=czibuildbot" | jq -r '.workflow_runs[] | select(.display_title == "stage") | .conclusion' | head -n 1)
+          if [ "$checks" == "success" ]; then
+            echo "All checks passed!"
+          else
+            echo "Some checks failed." 
+            exit 1
+          fi
+        shell: bash
+
+  merge_staging_to_prod:
+    runs-on: ubuntu-latest
+    needs:
+      - check_staging_commit_checks
+    steps:
+      - name: Checkout prod branch
+        uses: actions/checkout@v2
+        with:
+          ref: prod
+          fetch-depth: 0 # fetch all history to make merges work correctly
+      - name: Merge staging branch into prod branch
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "actions@github.com"
+
+          echo "Checking out 'prod' branch"
+          git checkout prod
+          echo "Confirming checked out branch receiving merge is: $(git branch --show-current)"
+
+          echo "Latest commit on 'prod' branch is: $(git log -1 --format='%H' prod)"
+          echo "Latest commit on 'staging' branch is: $(git log -1 --format='%H' staging)"
+
+          echo "About to merge 'staging' branch into 'prod' branch"
+          git merge --verbose staging -m "Merging staging branch into prod branch"
+
+          git push origin prod
+
+      # We need to use repository_dispatch because by default github actions
+      # prevents actions triggered by 'GITHUB_TOKEN' from triggering other
+      # workflows. Details are here:
+      # https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow
+      #https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch
+      - name: Trigger repository_dispatch to build images for prod
+        if: success()
+        uses: peter-evans/repository-dispatch@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          event-type: build-images-for-prod
+          client-payload: '{"ref": "refs/heads/prod"}'
+
+      - name: Send slack notification if staging not merged into prod
+        if: failure()
+        uses: 8398a7/action-slack@v3
+        with:
+          status: ${{ job.status }}
+          fields: repo,commit,author,eventName,workflow,job,mention
+          mention: "subteam^S02KZL4SQM6"
```

Comment:
who does this notify and where do you find this magic?

Reply:
it notifies the sc-oncall-eng slack group, and I got it from Joyce's work in [this](https://github.com/chanzuckerberg/single-cell-data-portal/pull/6413) PR updating the slack notifications